_posts/2022-01-10-reassign-partitions.md

```diff
@@ -0,0 +1,315 @@
+
+```sh
+kubectl cp config.properties <interactive_pod_name>:/tmp/config.properties
+```
+
+Lets create a `topics.json` file now. As we discussed above this file will have the topics that we need to reassign:
+
+```json
+{
+  "version": 1,
+  "topics": [
+    { "topic": "my-topic-1"},
+    { "topic": "my-topic-2"}
+  ]
+}
+```
+
+After the creation of this file, copy it to the interactive pod container since we will be running all of our commands from there:
+
+```sh
+kubectl cp topics.json <interactive_pod_name>:/tmp/topics.json
+```
+
+We can now start a shell process in our interactive pod container and run the command inside it to generate our proposal `reassignment.json` data.
+Let us start the shell process:
+
+```sh
+kubectl exec -ti <interactive_pod_name> /bin/bash
+```
+
+Now it's time for generating the `reassignment.json` file. 
+We will use the `kafka-reassign-partitions.sh` command to generate our proposal `reassignment.json` data:
+
+```sh
+bin/kafka-reassign-partitions.sh --bootstrap-server my-cluster-kafka-bootstrap:9093 \
+--command-config /tmp/config.properties \
+--topics-to-move-json-file /tmp/topics.json \
+--broker-list 0,1,2 \
+--generate
+```
+
+Once you run this command, you will be able to see the JSON data which is generated by the partition reassignment tool.
+Now you can copy and save this data to use in the `reassignment.json` file. You can also create/alter the `reassignment.json` file
+
+### Scaling down the cluster
+
+Coming to the last step of this example, Here we will see how we use the generated `reassignment.json` to get the reassignment done by the reassignment partition tool
+
+Since we have generated the `reassignment.json` file in the previous step, now we can use it for our next step.
+
+Lets copy the `reassignment.json` file to the interactive pod container.
+
+```sh
+kubectl cp reassignment.json <interactive_pod_name>:/tmp/reassignment.json
+```
+
+Now we will start a shell process inside the interactive pod container in which we will run our kafka bin script.
+```sh
+kubectl exec -ti <interactive_pod_name> /bin/bash
+```
+
+So lets run the `kafka-reassign-partitions.sh` script now to start the partition reassignment 
+
+```sh
+bin/kafka-reassign-partitions.sh --bootstrap-server
+ <cluster_name>-kafka-bootstrap:9093 \
+ --command-config /tmp/config.properties \
+ --reassignment-json-file /tmp/reassignment.json \ 
+ --execute
+```
+
+Now you can use the `--verify` mode to check if the partition reassignment is done or if it is still running.
+
+```sh
+bin/kafka-reassign-partitions.sh --bootstrap-server
+  <cluster_name>-kafka-bootstrap:9093 \
+  --command-config /tmp/config.properties \
+  --reassignment-json-file /tmp/reassignment.json \
+  --verify
+```
+The `--verify` mode reports if the assignment is done or not.
+
+Once the partition reassignment is complete the broker which were required to be scaled down will have no assigned partition and can be removed safely.
```

Comment:
In Strimzi, you can always scale down the last broker only. I think this should be reflected through the blog post and you should describe it more like a real scenario ... _I have 5 brokers and wanna scale to 3 because they are not too busy. How do I do it?_